.github/copilot-instructions.md

```diff
@@ -0,0 +1,204 @@
+# GitHub Copilot Instructions for IoTempower
+
+## Repository Overview
+
+IoTempower is an IoT framework for rapid deployment and management of wireless sensor/actuator networks using ESP8266/ESP32 microcontrollers. It provides device drivers, MQTT integration, OTA updates, and a configuration-based deployment system.
+
+## Key Architecture Concepts
+
+### System Hierarchy
+```
+IoT System (e.g., ~/iot-systems/my-home/)
+├── system.conf              # System-wide configuration (WiFi, MQTT broker)
+└── node-name/               # Each microcontroller has its own folder
+    ├── node.conf            # Board type and node-specific settings
+    ├── setup.cpp            # Device configuration (sensors/actuators)
+    ├── key.txt              # Security key for OTA updates
+    └── build/               # Auto-generated PlatformIO project (transient)
+```
+
+**Important**: Nodes and devices are defined declaratively in `setup.cpp` using a high-level C++ DSL. The system automatically handles MQTT topics, networking, security, and OTA updates.
+
+## Directory Structure
+
+### Documentation (`/doc`)
+- `architecture.rst` - System architecture overview
+- `first-node.rst` / `second-node.rst` - Essential tutorials
+- `device-architecture.rst` - Device framework details
+- `deployment-process.rst` - Build and deployment workflow
+- `node_help/` - Device command references
+- `projects_help/` - Project configuration help
+
+### System Management (`/bin`)
+- `compile` - Compiles node firmware
+- `deploy` - Deploys firmware (serial or OTA)
+- `command_checker.py` - Detects which device drivers are needed
+- `create_node_template` - Creates new node folders
+- Other gateway management scripts (accesspoint, MQTT, etc.)
+
+### MCU Device Drivers (`/lib/node_types/esp/src`)
```

Comment:
You explain the explicit inheritance (that is worth mentioning here), I was more thinking of the implicit inheritance that esp32 is overwriting files through its base symbolic link from the esp folder, while m5stickc is overwriting files from esp32 and esp.. Please add an explanation about this.